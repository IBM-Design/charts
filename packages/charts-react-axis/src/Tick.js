import React, { PropTypes } from 'react';

const Tick = (props) => {
  const { dy, label, offset, path, x, y } = props;

  return (
    <g className="tick" transform={offset}>
      <path
        d={path}
        fill="none"
        stroke="#CDDBF1"
        strokeWidth="1"
      />
      <text fill="#000" x={x} y={y} dy={`${dy}em`}>
        {label}
      </text>
    </g>
  );
};

Tick.propTypes = {
  dy: PropTypes.number.isRequired,
  label: PropTypes.string.isRequired,
  offset: PropTypes.string.isRequired,
  path: PropTypes.string.isRequired,
  x: PropTypes.number.isRequired,
  y: PropTypes.number.isRequired,
};

export default Tick;
