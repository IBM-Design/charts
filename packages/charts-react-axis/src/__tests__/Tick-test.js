import React from 'react';
import renderer from 'react-test-renderer';
import Tick from '../Tick';

describe('Tick Component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <Tick
        dy={0.71}
        height={500}
        label="Label"
        offset="translate(0, 0)"
        path="M0,0 L0,-500"
        x={0.5}
        y={9}
      />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
