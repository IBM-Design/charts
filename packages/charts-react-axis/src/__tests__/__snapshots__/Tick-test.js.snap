exports[`Tick Component should render 1`] = `
<g
  className="tick"
  transform="translate(0, 0)">
  <path
    d="M0,0 L0,-500"
    fill="none"
    stroke="#CDDBF1"
    strokeWidth="1"
    style={
      Object {
        "animation": undefined,
        "strokeDasharray": undefined,
        "strokeDashoffset": undefined,
      }
    } />
  <text
    dy="0.71em"
    fill="#000"
    x={0.5}
    y={9}>
    Label
  </text>
</g>
`;
